lib/LANraragi/Utils/Logging.pm

```diff
@@ -31,89 +46,105 @@ sub get_logdir {
     return $log_folder;
 }
 
-# Returns a Logger object with a custom name and a filename for the log file.
+# Returns a Mojo::Log object with a custom name and a filename for the log file.
 sub get_logger {
 
     #Customize log file location and minimum log level
     my $pgname  = $_[0];
     my $logfile = $_[1];
 
-    my $logpath = get_logdir . "/$logfile.log";
-
-    if ( -e $logpath && -s $logpath > 1048576 ) {
-
-        # Rotate log if it's > 1MB
-        say "Rotating logfile $logfile";
+    my $logpath     = get_logdir . "/$logfile.log";
+    my $cache_key   = "$logfile";
+    my $log;
+
+    # Reuse cached logger if exists, otherwise clean cache and recreate
+    my $cache_refresh_error;
+    if ( exists $LOGGER_CACHE{$cache_key} && -e $logpath ) {
+        $log = $LOGGER_CACHE{$cache_key};
+
+        my $ok;
+        {
+            local $@;
+            $ok = eval {
+                LANraragi::Utils::RotatingLog::refresh_logger_handle($log);
+                1;
+            };
+            $cache_refresh_error = $@ unless $ok;
+        }
 
-        # Based on Logfile::Rotate
-        # Rotate existing logs
-        for ( my $i = 7; $i > 1; $i-- ) {
-            my $j = $i - 1;
-            my $next = "$logpath.$i.gz";
-            my $prev = "$logpath.$j.gz";
-            if ( -r $prev && -f $prev ) {
-                rename( $prev, $next ) or die "error: rename failed: ($prev,$next)";
+        if ($ok) {
```

Comment:
IMO this can be flattened a bit, I don't think you really need `$ok` (or the localized error context but it _might_ pollute the later loop so might as well leave it in)  

Using `experimental::try` might also be a valid approach here? 🤔 

```suggestion
        {
            local $@;
            eval {
                    LANraragi::Utils::RotatingLog::refresh_logger_handle($log); 
            };
        
            $cache_refresh_error = $@;
        }
        
        unless ($cache_refresh_error) {
```